File: Factorial.java
import java.util.Scanner;

/**
 * Class-example of factorial implementation
 * 
 * @author VitasSalvantes
 * @version 2.0
 */
public class Factorial {

    /**
     * Method that launches the program
     */
    public static void main(String[] args) {
        int result = 1;
        System.out.println("- Number?");
        int number = new Scanner(System.in).nextInt();
        System.out.print(number + "! = ");

        while (number > 0) {
            result *= number;
            number--;
        }

        System.out.println(result);
    }
}

File: Factorial.kt
/**
 * Example of factorial implementation
 *
 * @author VitasSalvantes
 * @version 1.0
 */
fun main() {
    var result: Int = 1
    println("- Number?")
    var number: Int = readLine()?.toInt() ?: 0
    print("$number! = ")

    while (number > 0) {
        result *= number
        number = number.dec()
    }

    println(result)
}